### PR TITLE
Update User Guide for argument clarity for Index based Commands

### PR DESCRIPTION
**Pull request by colinhia**
Following the discussion in #308, brought out into a separate PR.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -480,6 +482,8 @@ All client information, including contact details, address, job information, and
 edit <INDEX> [n/ <NAME>] [p/ <PHONE>] [e/ <EMAIL>] [a/ <ADDRESS>] [j/ <JOB>] [i/ <INCOME>] [t/ <TIER>] [rn/ <NEW REMARK>] [ra/ <REMARK TO BE APPENDED>] [s/ <STATUS>]
 ```
 - Mandatory Field: `<INDEX>`, at least one of the Optional Fields
+  - **Note:** The provided index must be **greater than 0**, and **less than the total number of clients in the list**.
```
Comment (FionaQY) [inline docs/UserGuide.md:485]: I think this should be "less than or equals to" for clarity's sake.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -549,6 +553,8 @@ For detailed explanations of each flag and acceptable arguments, refer to Sectio
 delete <INDEX>
 ```
 * Mandatory Field: `<INDEX>`
+  - **Note:** The provided index must be **greater than 0**, and **less than the total number of clients in the list**.
```
Comment (FionaQY) [inline docs/UserGuide.md:556]: ditto

File: docs/UserGuide.md [hunk 3]
```diff
@@ -730,6 +736,8 @@ For detailed explanations for the matching criteria of each flag and the accepta
 view <INDEX>
 ```
 * Mandatory Field: `<INDEX>`
+  - **Note:** The provided index must be **greater than 0**, and **less than the total number of clients in the list**.
```
Comment (FionaQY) [inline docs/UserGuide.md:739]: dittoditto

**Review by FionaQY — APPROVED** [review]
LGTM. Just a few nits

File: docs/UserGuide.md (hunk 3 above)
Comment (colinhia) [reply docs/UserGuide.md:739]: Thanks for the catch!

**Review by ZShunRen — APPROVED** [review]
LGTM

**Pull request merged**